aten/src/ATen/native/cuda/RowwiseScaledMM.cu

```diff
@@ -0,0 +1,536 @@
+#define TORCH_ASSERT_ONLY_METHOD_OPERATORS
+#include <ATen/Dispatch.h>
+#include <ATen/core/Tensor.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/cuda/nvrtc_stub/ATenNVRTC.h>
+
+// Determine if the architecture supports rowwise scaled mm
+#if !defined(USE_ROCM) && defined(CUDA_VERSION) && CUDA_VERSION >= 12000
+
+#define BUILD_ROWWISE_FP8_KERNEL
+#endif
+
+#if defined(BUILD_ROWWISE_FP8_KERNEL)
+
+// We are going to override the cuTensorMapEncodeTiled driver api with our lazy loader
+static CUresult CUDAAPI nvrtc_cuTensorMapEncodeTiled(
+    CUtensorMap* tensorMap,
+    CUtensorMapDataType tensorDataType,
+    cuuint32_t tensorRank,
+    void* globalAddress,
+    const cuuint64_t* globalDim,
+    const cuuint64_t* globalStrides,
+    const cuuint32_t* boxDim,
+    const cuuint32_t* elementStrides,
+    CUtensorMapInterleave interleave,
+    CUtensorMapSwizzle swizzle,
+    CUtensorMapL2promotion l2Promotion,
+    CUtensorMapFloatOOBfill oobFill) {
+  return at::globalContext().getNVRTC().cuTensorMapEncodeTiled(
+      tensorMap,
+      tensorDataType,
+      tensorRank,
+      globalAddress,
+      globalDim,
+      globalStrides,
+      boxDim,
+      elementStrides,
+      interleave,
+      swizzle,
+      l2Promotion,
+      oobFill);
+}
+
+// Rename the global function symbol
+#define cuTensorMapEncodeTiled nvrtc_cuTensorMapEncodeTiled
```

Comment:
Thanks for the comment! I did not know about: [CudaHostAdapter](https://github.com/NVIDIA/cutlass/blob/637b15906358191cb4238af419d408a65819d7ec/include/cutlass/cuda_host_adapter.hpp#L66-L68). This does in fact feel like the "correct" solution.

I wonder if its okay to have this hack temporarily until we can upstream the change to cutlass

cc @malfet

Reply:
I think it's ok to open an issue on CUTLASS to ask them for a better API, and implement a workaround in the meantime.

Personally, I find the current workaround acceptable (I can't find any concrete issue with it).

One concern, perhaps, is that the preprocessor-driven substitution might be overly broad. For example, if one of the headers transitively includes `cuda.h`, then we would be changing the definition in there as well. This might be find, but seems intrusive. A more targeted solution might be to only apply the substitution when importing the one header file that uses this function, namely [`cute/atom/copy_traits_sm90_tma.hpp`](https://github.com/NVIDIA/cutlass/blob/637b15906358191cb4238af419d408a65819d7ec/include/cute/atom/copy_traits_sm90_tma.hpp#L922). It would look this this:
```cpp
// Don't replace the definition in the CUDA driver headers
#include <cuda.h>

// Replace the usage inside CUTLASS
#define cuTensorMapEncodeTiled nvrtc_cuTensorMapEncodeTiled
#include <cute/atom/copy_traits_sm90_tma.hpp>
#undef cuTensorMapEncodeTiled

// Import the rest of CUTLASS as usual (this won't reimport the above header thanks to pragma once)
#include <cutlass/cutlass.h>
```

Up to you to decide if the added complexity is worth the reduced risk.

Reply:
Totally agree! let me limit the blast radius and make an issue on cutlass